### PR TITLE
added the adapters adapter

### PR DESCRIPTION
**Pull request by yjv**
This adds a new adapter that allows you to put multiple adapters in it and paginate through them all. Useful for multiple datasources

### CODE REVIEW

File: src/Pagerfanta/Adapter/AdaptersAdapter.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+class AdaptersAdapter implements AdapterInterface
+{
+    /**
+     * @var AdapterInterface[]
+     */
+    protected $adapters;
+
+    /**
+     * AdaptersAdapter constructor.
+     * @param $adapters
```
Comment (Koc) [inline src/Pagerfanta/Adapter/AdaptersAdapter.php]: `@param AdapterInterface[] $adapters`

**Comment by yjv** [comment]
@Koc is this good to be merged?

thanks

**Comment by Koc** [comment]
:+1: from me, but I haven't right to merge anything to this repo.

ping @richsage @pablodip

**Comment by Koc** [comment]
also you can squash the commits

**Comment by yjv** [comment]
@Koc @richsage @pablodip commits have been squashed.

Thanks

**Comment by yjv** [comment]
@Koc @richsage @pablodip wondering whats happening with this.

Thanks!

File: README.md [hunk 2]
```diff
@@ -333,6 +333,38 @@ $results = array(/* ... */);
 $adapter = new FixedAdapter($nbResults, $results);
 ```
 
+### AdaptersAdapter
+
+This adapter can take an array of any other adapters and go through them
```
Comment (pablodip) [inline README.md]: Could you possibly explain further what this adapter does? How does it treat both the number of results and the slice with several adapter?

Comment (stof) [inline README.md]: Reading the code, this is equivalent to paginating an AppendIterator containing the different iterators being wrapped in adapters

File: src/Pagerfanta/Adapter/AdaptersAdapter.php [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+ *
+ * (c) Pablo Díez <pablodip@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Pagerfanta\Adapter;
+
+class AdaptersAdapter implements AdapterInterface
```
Comment (stof) [inline src/Pagerfanta/Adapter/AdaptersAdapter.php]: The name of the class is a bad name, as it does not describe how the adapters are combined. I can think of at least 2 ways to implement an adapter bring other adapters together:

- considering that it paginates an AppendIterator of the different iterators (what you implement it)
- considering that it paginates a MultipleIterator looping over the different iterators in parallel

**Comment by FinesseRus** [comment]
`AdaptersAdapter` does the same as this adapter: https://github.com/whiteoctober/Pagerfanta/blob/master/src/Pagerfanta/Adapter/ConcatenationAdapter.php

**Pull request closed without merging**